modules/standard/FileSystem.chpl

```diff
@@ -1114,6 +1114,59 @@ proc remove(name: string) {
   if err != ENOERR then ioerror(err, "in remove", name);
 }
 
+private proc rmTreeHelper(out error: syserr, root: string) {
+  // Go through all the files in this current directory and remove them
+  for filename in listdir(path=root, dirs=false, files=true, listlinks=true) {
+    var name = root + "/" + filename;
+    remove(error, name);
+    if (error != ENOERR) then return;
+  }
+  // Then traverse all the directories within this current directory and have
+  // them handle cleaning up their contents and themselves
+  for dirname in listdir(path=root, dirs=true, files=false, listlinks=false) {
+    rmTreeHelper(error, root+"/"+dirname);
+    if (error != ENOERR) then return;
+  }
+  // Once everything else has been removed, remove ourself.
+  remove(error, root);
+}
+
+pragma "no doc"
+proc rmTree(out error: syserr, root: string) {
+  var expectedErrorCases = !exists(error, root);
+  if (error != ENOERR) then return; // Some error occurred in checking the existence of root.
+  else if (expectedErrorCases) {
```

Comment:
I find the variable name "expectedErrorCases" confusing.
I'd prefer different variables like "rootMissing" or
"rootNotDir".